src/EVMBlackjack.sol

```diff
@@ -0,0 +1,337 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.16;
+
+import {Chip} from "../src/Chip.sol";
+
+contract EVMBlackjack {
+    /*//////////////////////////////////////////////////////////////
+    //  Events
+    //////////////////////////////////////////////////////////////*/
+
+    event BetPlaced(address indexed player, uint16 betSize);
+    event PlayerCardDealt(address indexed player, uint8 card, uint8 handIndex);
+    event DealerCardDealt(address indexed player, uint8 card);
+    event InsuranceTaken(address indexed player, bool take);
+    event PlayerActionTaken(address indexed player, Action action);
+    event PlayerBust(address indexed player);
+    event DealerBust(address indexed player);
+    event PayoutsHandled(address indexed player, uint16 playerPayout, uint16 dealerPayout);
+
+    /*//////////////////////////////////////////////////////////////
+    //  Errors
+    //////////////////////////////////////////////////////////////*/
+
+    error InvalidStateTransition();
+    error InvalidBetSize(uint16 betSize);
+
+    /*//////////////////////////////////////////////////////////////
+    //  Data Structures
+    //////////////////////////////////////////////////////////////*/
+
+    enum State {
+        READY_FOR_BET,
+        WAITING_FOR_RANDOMNESS,
+        READY_FOR_INSURANCE,
+        READY_FOR_PLAYER_ACTION,
+        DEALER_ACTION,
+        PAYOUTS
+    }
+
+    enum Action {
+        NO_ACTION,
+        SPLIT_ACES,
+        SPLIT,
+        DOUBLE_DOWN,
+        HIT,
+        STAND
+    }
+
+    struct Game {
+        State state;
+        uint16 shoeCount;
+        uint8[] dealerCards;
+        uint16 insurance;
+        Action lastAction;
+        uint8 totalPlayerHands;
+        uint8 activePlayerHand;
+        Hand[] playerHands;
+    }
+
+    // uint8[] playerCards;
+    // uint16 betSize;
+
+    struct Hand {
+        uint8[] cards;
+        uint16 betSize;
+    }
+
+    /*//////////////////////////////////////////////////////////////
+    //  Private Variables -- State
+    //////////////////////////////////////////////////////////////*/
+
+    Chip internal chip;
+
+    mapping(address => Game) internal games;
+
+    /*//////////////////////////////////////////////////////////////
+    //  Private Variables -- Constant
+    //////////////////////////////////////////////////////////////*/
+
+    uint16 internal constant MINIMUM_BET_SIZE = 10;
```

Comment:
is this in CHIP?

Reply:
Yes, again with decimals of 0, as of now

Reply:
Resolved, 18 decimals now